### DISCUSSION TITLE
How to access task schedular from IIS apppoll when IIS has no admin access

### DISCUSSION THREAD

**Discussion by achyutghosh** [dahall/TaskScheduler#957] [Q&A]
Hi,

I want to build a web API application (of course in c#) that will be hosted in IIS. The responsibility of this application will be to fetch the task scheduler status, but since the IIS has no admin permission, it is not returning any scheduler. So is there any way to get this?

**Comment by dahall** [comment]
Do the notes at #859 or #494 help?

**Comment by kedar-hatkar** [comment]
I am using below example to update task scheduler time from webapi hosted on IIS. Provide valid user credentials having admin rights

using (TaskService taskservice = new TaskService("MachineName", "UserName", "DomainName", "Password"))
{
var TaskName = "abcd";
var task = taskservice.GetTask(TaskName);
if (task != null)
{
var td = task.Definition;
td.Triggers[0].Repetition.Interval = TimeSpan.FromMinutes(45);
taskservice.RootFolder.RegisterTaskDefinition(TaskName, td, TaskCreation.Update, UserName);
///log Task Updated successfully
}
else
{
/// task doesn't exist in task scheduler"
}
}
